code/__DEFINES/dcs/signals/signals_object.dm

```diff
@@ -246,6 +246,8 @@
 
 /// Called on component/uplink/OnAttackBy(..)
 #define COMSIG_ITEM_ATTEMPT_TC_REIMBURSE "item_attempt_tc_reimburse"
+///Called when a holoparasite/guardiancreator is used. 
+#define COMSIG_TRAITOR_ITEM_USED(/obj/item/guardiancreator) "traitor_item_used_[/obj/item/guardiancreator]"
```

Comment:
Oh no, I don't think I explained it well. Here's a mockup of how the system should flow.

```dm
#define COMSIG_TRAITOR_ITEM_USED(type) "traitor_item_used_[type]"

/datum/element/uplink_reimburse/Attach(datum/target, refundable_tc = 1)
 . = ..()
 RegisterSignal(target, COMSIG_TRAITOR_ITEM_USED(target.type), PROC_REF(on_use)) // or used() as you have it in current code

/obj/item/guardiancreator/attack_self()
 // whatever code goes here
 SEND_SIGNAL(src, COMSIG_TRAITOR_ITEM_USED(type))
```

This should make your code compile, and best of all, make it the system I was envisioning. The reason why we rely on `type` so much is because it is immutable on runtime (as opposed to the name or something, which can obviously be changed by an admin for some reason).

Reply:
Oh I see. Ill fix it up then.

Reply:
Alrighty, made the changes. Should be good!